### ISSUE TITLE
Build failure against qt 5.15

### ISSUE THREAD

**Issue by fmauch** [ros-noetic-arch/ros-noetic-qt-gui-cpp#1]
Installing this on an up-to-date system seems to fail due to changed Qt API.

```
/home/mauch/.cache/yay/ros-noetic-qt-gui-cpp/src/build/sip/qt_gui_cpp_sip/siplibqt_gui_cpp_sipqt_gui_cppPluginContext.cpp: In function ‘void assign_qt_gui_cpp_PluginContext(void*, Py_ssize_t, void*)’:
/home/mauch/.cache/yay/ros-noetic-qt-gui-cpp/src/build/sip/qt_gui_cpp_sip/siplibqt_gui_cpp_sipqt_gui_cppPluginContext.cpp:177:131: error: use of deleted function ‘qt_gui_cpp::PluginContext& qt_gui_cpp::PluginContext::operator=(const qt_gui_cpp::PluginContext&)’
  177 |     reinterpret_cast< ::qt_gui_cpp::PluginContext *>(sipDst)[sipDstIdx] = *reinterpret_cast< ::qt_gui_cpp::PluginContext *>(sipSrc);
      |                                                                                                                                   ^
In file included from plugin_context.sip:7:
/home/mauch/.cache/yay/ros-noetic-qt-gui-cpp/src/qt_gui_core-0.4.0/qt_gui_cpp/src/qt_gui_cpp_sip/../../include/qt_gui_cpp/plugin_context.h:53:23: note: ‘qt_gui_cpp::PluginContext& qt_gui_cpp::PluginContext::operator=(const qt_gui_cpp::PluginContext&)’ is implicitly deleted because the default definition would be ill-formed:
   53 | class QT_GUI_CPP_DECL PluginContext
      |                       ^~~~~~~~~~~~~
/home/mauch/.cache/yay/ros-noetic-qt-gui-cpp/src/qt_gui_core-0.4.0/qt_gui_cpp/src/qt_gui_cpp_sip/../../include/qt_gui_cpp/plugin_context.h:53:23: error: use of deleted function ‘QObject& QObject::operator=(const QObject&)’
In file included from /usr/include/qt/QtCore/qmetatype.h:45,
                 from /usr/include/qt/QtCore/QMetaType:1,
                 from sipAPIlibqt_gui_cpp_sip.h:12,
                 from siplibqt_gui_cpp_sipqt_gui_cppPluginContext.cpp:7:
/usr/include/qt/QtCore/qglobal.h:434:12: note: declared here
  434 |     Class &operator=(const Class &) = delete;
      |            ^~~~~~~~
/usr/include/qt/QtCore/qobject.h:467:5: note: in expansion of macro ‘Q_DISABLE_COPY’
  467 |     Q_DISABLE_COPY(QObject)
      |     ^~~~~~~~~~~~~~
make[3]: *** [Makefile:18: siplibqt_gui_cpp_sipqt_gui_cppPluginContext.o] Error 1
```

I've been digging around just a little bit and it seems to boil down to this python-sip stuff that I don't really understand atm. The generated code seems to use an operator it shouldn't be using, so I guess, the problem is lying upstream with python-sip? But that's just a quick guess.

**Comment by acxz** [comment]
Yeah it most likely is due to `sip` the recent upgrade from `4.19.22` to `4.19.23` broke some things, including other packages like, https://github.com/ros-noetic-arch/ros-noetic-python-orocos-kdl/issues/1 (see associated upstream issue).
I tried to muddle through the changelog here https://www.riverbankcomputing.com/hg/sip but didn't really have the time to dig deeper.

**Comment by acxz** [comment]
If you could report this upstream and link the corresponding issue back here that would be great.

**Comment by acxz** [comment]
Went ahead and patched it, keeping this issue open until the patch is included in a new release.

**Comment by acxz** [comment]
fixed with `0.4.2-1` closing issue
